pulsar-broker/src/main/java/org/apache/pulsar/broker/service/plugin/EntryFilterProvider.java

```diff
@@ -0,0 +1,149 @@
+                    checkArgument(StringUtils.isNotBlank(def.getEntryFilterClass()));
+
+                    EntryFilterMetaData metadata = new EntryFilterMetaData();
+                    metadata.setDefinition(def);
+                    metadata.setArchivePath(archive);
+
+                    entryFilterDefinitions.getFilters().put(def.getName(), metadata);
+                } catch (Throwable t) {
+                    log.warn("Failed to load entry filters from {}."
+                            + " It is OK however if you want to use this entry filters,"
+                            + " please make sure you put the correct entry filter NAR"
+                            + " package in the entry filter directory.", archive, t);
+                }
+            }
+        }
+
+        return entryFilterDefinitions;
+    }
+
+    private static EntryFilterDefinition getEntryFilterDefinition(String narPath,
+                                                                              String narExtractionDirectory)
+            throws IOException {
+        try (NarClassLoader ncl = NarClassLoader.getFromArchive(new File(narPath), Collections.emptySet(),
+                narExtractionDirectory)) {
+            return getEntryFilterDefinition(ncl);
+        }
+    }
+
+    private static EntryFilterDefinition getEntryFilterDefinition(NarClassLoader ncl) throws IOException {
+        String configStr = ncl.getServiceDefinition(ENTRY_FILTER_DEFINITION_FILE);
+
+        return ObjectMapperFactory.getThreadLocalYaml().readValue(
+                configStr, EntryFilterDefinition.class
+        );
+    }
+
+    private static EntryFilterWithClassLoader load(EntryFilterMetaData metadata,
+                                                               String narExtractionDirectory)
+            throws IOException {
+        NarClassLoader ncl = NarClassLoader.getFromArchive(
+                metadata.getArchivePath().toAbsolutePath().toFile(),
+                Collections.emptySet(),
+                BrokerInterceptor.class.getClassLoader(), narExtractionDirectory);
+
+        EntryFilterDefinition def = getEntryFilterDefinition(ncl);
+        if (StringUtils.isBlank(def.getEntryFilterClass())) {
+            throw new IOException("Entry filters `" + def.getName() + "` does NOT provide a broker"
+                    + " interceptors implementation");
+        }
+
+        try {
+            Class entryFilterClass = ncl.loadClass(def.getEntryFilterClass());
+            Object filter = entryFilterClass.getDeclaredConstructor().newInstance();
+            if (!(filter instanceof EntryFilter)) {
+                throw new IOException("Class " + def.getEntryFilterClass()
+                        + " does not implement entry filter interface");
+            }
+            EntryFilter pi = (EntryFilter) filter;
+            return new EntryFilterWithClassLoader(pi, ncl);
+        } catch (Throwable t) {
+            return null;
```

Comment:
Please log the error and fail 
If the user configured a filter then we must load it
If the broker starts without the filter we will have an unexpected behaviour, like messages that are dispatched without permission